File: tools/phylogeny.fr-v2/conversion/fastatophylip.pl
#!/usr/bin/perl

# This a shell wrapper to convert MSA format: FASTA <-> PHYLIP
#
# Author: C. Vernette, V. Lefort
# Date: June 2014
# Version: 1.0
#

#The arguments are: the input file; the output file; the choice of the user

use strict;
use warnings;
use Bio::AlignIO;

my $id=0;
open (FILE,$ARGV[0]) || die "impossible d'ouvrir le fichier";

#Convert a file from FASTA to PHYLIP

if ($ARGV[2] eq "phylip"){
	my $in=Bio::AlignIO->new(-file=> $ARGV[0] ,-format=>'fasta');
	while (<FILE>){
		if( /^>(.+)/ ){
			my $ide= length($1);
			if($ide>$id){
				$id=$ide;
			}
		}
	}
	$id=$id+5;
	my $out=Bio::AlignIO->new(-file=> ">>$ARGV[1]",-format=>'phylip',-idlength=>$id);
	while(my $aln=$in->next_aln){
		$out->write_aln($aln);
	}
}

#Convert a file from PHYLIP to FASTA

if ($ARGV[2] eq "fasta"){
	open(OUT, ">>$ARGV[1]") || die "Can't open out $ARGV[1]: $!\n";
	while(my $line = <FILE>) {
		chomp $line;
		next if (!$line || $line =~ /\d+\s+\d+/);
		if($line =~ /^(.*)\s+(\S+)\s*$/) {
			my $name = $1;
			my $sequence = $2;
			$name =~ s/\s+$//;
			print OUT ">$name\n$sequence\n";
		}
	
	}
	close OUT;
}

close FILE;

File: tools/phylogeny.fr-v2/conversion/fasta_to_phylip.py
# -*- coding: utf-8 -*-
#!/usr/bin/env python
from Bio import AlignIO

from optparse import OptionParser


# main
def main():
    # define options
    parser = OptionParser()
    parser.add_option("-i", "--input")
    parser.add_option("-o", "--output")

    # parse
    options, args = parser.parse_args()

    # retrieve options
    inputFasta = options.input
    outputPhylip = options.output
    input_handle = open(inputFasta, "rU")
    output_handle = open(outputPhylip, "w")

    alignments = AlignIO.parse(input_handle, "fasta")
    AlignIO.write(alignments, output_handle, "phylip")

    output_handle.close()
    input_handle.close()

        
if __name__ == "__main__":
    main()
    